### PR TITLE
Reset home position when landed and disarmed

### PR DESCRIPTION
**Pull request by sanderux**
As discussed in https://github.com/PX4/Firmware/issues/8468#issuecomment-352581115

Reset home position when landed and disarmed

### CODE REVIEW

**Comment by priseborough** [comment]
How have you tested this?

**Comment by sanderux** [comment]
Sitl

**Comment by priseborough** [comment]
Please provide a link to the log.

**Comment by sanderux** [comment]
let me re-run with logging while disarmed, or you wont see anything change

File: msg/home_position.msg [hunk 1]
```diff
@@ -15,3 +15,4 @@ float32 direction_z		# Takeoff direction in NED Z
 
 bool valid_alt		# true when the altitude has been set
 bool valid_hpos		# true when the latitude and longitude have been set
+bool manual_home	# true when home position was set manually
```
Comment (mhkabir) [inline msg/home_position.msg]: I don't see this field being published anywhere. What is the plan here?

**Comment by sanderux** [comment]
@mhkabir ah good catch, i had that done and run astyle over commander. after that bad idea i rewinded and reimplemented. Let me re-add that

**Comment by sanderux** [comment]
@priseborough sit doesnt seem to publish SDLOG_MODE param.
not sure how to log after disarmed, any hints?

**Comment by sanderux** [comment]
Log of flight while armed:
https://logs.px4.io/plot_app?log=0237dfda-5a9f-47eb-97c7-0d04df15d1f2

And here's some mavlink data
image

image

moved land wp a little to show home:
image

File: src/modules/commander/commander.cpp [hunk 2]
```diff
@@ -3120,18 +3122,31 @@ int commander_thread_main(int argc, char *argv[])
 		/* Get current timestamp */
 		const hrt_abstime now = hrt_absolute_time();
 
+		/* distance from home */
+
+		float home_dist_xy = -1.0f;
+		float home_dist_z = -1.0f;
+		get_distance_to_point_global_wgs84(_home.lat, _home.lon, _home.alt, global_position.lat, global_position.lon,
```
Comment (dagar) [inline src/modules/commander/commander.cpp]: Do this in the local frame.

**Comment by sanderux** [comment]
SITL tested and functions as expected.
Also tested with a manual home position and this also works correctly:

image

**Comment by sanderux** [comment]
Also tested after manually setting home and then resetting.
All works

**Comment by dagar** [comment]
I also did a bit of SITL testing last night. https://review.px4.io/plot_app?log=ece4698c-993d-4ea9-95f4-2704d9416b09

image

File: src/modules/commander/commander.cpp [hunk 3]
```diff
@@ -3120,24 +3135,43 @@ int commander_thread_main(int argc, char *argv[])
...
-		/* First time home position update - but only if disarmed */
-		if (!status_flags.condition_home_position_valid && !armed.armed) {
-			commander_set_home_position(home_pub, _home, local_position, global_position, attitude, false);
-		}
-
-		/* update home position on arming if at least 500 ms from commander start spent to avoid setting home on in-air restart */
-		else if (((!was_armed && armed.armed) || (was_landed && !land_detector.landed)) &&
-			(now > commander_boot_timestamp + INAIR_RESTART_HOLDOFF_INTERVAL)) {
-			commander_set_home_position(home_pub, _home, local_position, global_position, attitude, false);
+		// automaticcally set or update home position
```
Comment (priseborough) [inline src/modules/commander/commander.cpp]: 'automatically'

**Comment by sanderux** [comment]
Field tested: https://logs.px4.io/plot_app?log=d0b2ca5f-7baf-429f-bc19-afba65c806ce
Here you see the home_pos publishing a new alt right at disarm:
image

**Pull request merged**